integrations/apify/src/client.ts

```diff
@@ -0,0 +1,393 @@
+    startUrls: string[];
+    excludeUrlGlobs?: string[];
+    includeUrlGlobs?: string[];
+    maxCrawlPages?: number;
+    saveMarkdown?: boolean;
+    htmlTransformer?: 'readableText' | 'readableTextIfPossible' | 'minimal' | 'none';
+    removeElementsCssSelector?: string;
+    crawlerType?: 'playwright:adaptive' | 'playwright:firefox' | 'cheerio' | 'jsdom' | 'playwright:chrome';
+    expandClickableElements?: boolean;
+    headers?: Record<string, string>;
+    rawInputJsonOverride?: string;
+    kbId?: string;
+  }) {
+    try {
+      let input = params.rawInputJsonOverride ? (() => {
+        const parsed = JSON.parse(params.rawInputJsonOverride);
+        return parsed.startUrls && Array.isArray(parsed.startUrls) 
+          ? { ...parsed, startUrls: parsed.startUrls.map((url: string | { url: string }) => 
+              typeof url === 'string' ? { url } : url
+            ) }
+          : parsed;
+      })() : {
+        startUrls: params.startUrls.map(url => ({ url })),
+        excludeUrlGlobs: params.excludeUrlGlobs || [],
+        includeUrlGlobs: params.includeUrlGlobs || ['**/*'],
+        maxCrawlPages: params.maxCrawlPages || 10,
+        saveMarkdown: params.saveMarkdown ?? false, 
+        htmlTransformer: params.htmlTransformer || 'readableTextIfPossible',
+        removeElementsCssSelector: params.removeElementsCssSelector || '',
+        crawlerType: params.crawlerType || 'playwright:firefox',
+        expandClickableElements: params.expandClickableElements || false,
+        kbId: params.kbId || "kb-default",
+      };
+      
+      if (params.headers && Object.keys(params.headers).length > 0) {
+        input.headers = params.headers;
+      }
+
+      this.logger.forBot().info('Starting Apify Website Content Crawler with input:', JSON.stringify(input, null, 2));
+
+      const run = await this.client.actor('apify/website-content-crawler').call(input, { waitSecs: 0 });
+
+      if (!run || !run.id) {
+        throw new Error('Failed to start crawler run');
+      }
+
+      this.logger.forBot().info(`Crawler run started with ID: ${run.id}`);
+
+      return {
+        success: true,
+        message: 'Crawler run started successfully',
+        data: {
+          runId: run.id,
+          status: run.status,
+        },
+      };
+    } catch (error) {
+      this.logger.forBot().error('Error in startCrawlerRun:', error);
+      
+      if (error instanceof ApifyApiError) {
+        const { message, type, statusCode, clientMethod } = error;
+        this.logger.forBot().error('Apify API Error:', { message, type, statusCode, clientMethod });
+        
+        return {
+          success: false,
+          message: `Apify API Error: ${message}`,
+          data: {
+            error: message,
+            type,
+            statusCode,
+            clientMethod,
+          },
+        };
+      }
+
+      return {
+        success: false,
+        message: error instanceof Error ? error.message : 'Unknown error occurred',
```

Comment:
nitpick: if we had tighter try catches, we might reduce the amount of unknown errors we throw, because we can at least say "error during XYZing"